### DISCUSSION TITLE
How to post a JSON object with an array in it?

### DISCUSSION THREAD

**Discussion by Bouchtaoui** [pocketbase/pocketbase#1004] [Q&A] [answered]
I really love PocketBase and I'm so thankful to you :)

I have a Project object with a field 'members' containing an array of user ids. But in your documentation there's no way to define an array type field for a collection. Is this not possible?

How can I have a record with an array field?

I tried the following:

`
const collection = await client.collections.create({
			name: "projects",
			schema: [
				{
					name: "title",
					type: "text",
					required: true,
					options: {
						min: 3,
					},
				},
				{
					name: "description",
					type: "text",
				},
{
					name: "members",
					type: "array", // <-- is this even possible?,
				},
				{
					name: "color",
					type: "number",
					default: "blue",
				},
			],
		})
`

**Comment by ganigeorgiev** [comment, marked as answer]
@Bouchtaoui I'm not sure that I understand the question. Are you asking for creating a collection OR record? "Collection" is the structure of your data (aka. table), while "Record" represent a single entry in the collection.

PocketBase doesn't have dedicated "array" type of fields for the collections. More documentation will be available soon, but for a list with all supported fields you could check the table from https://github.com/pocketbase/pocketbase/releases/tag/v0.3.0.
From the above table you can see that array-like fields are the `select`, `relation` and `file` fields when the `options.maxSelect != 1` (`user` type is removed in v0.8.0-rc per #376).

Depending on what you are trying to do, you can create `select` or `relation` type fields with `options.maxSelect > 1` from the Admin UI and then submit a record with the array values:

```js
client.records.create("projects", {
    "members": ["optionA", "optionB"],
})
```

Could  you elaborate what you are trying to do?

**Reply by Bouchtaoui** [reply]
Thank you very much for your time to answer my question.
What I want to achieve, is to add users (as members) to my project to allow them to access my project.

So for each project in collection "projects" I want a field that lists user ids (of max 10 users). This way, only these users can view or update the project. So the project is like shared with other users, I don't know if it's possible. Forgive me for this simple question, I'm kind of new to backend logic, but I really like it.

**Reply by ganigeorgiev** [reply]
@Bouchtaoui Yes, this is possible. There are several ways you can approach this, but I think the easiest way would be:

1. Create a new `projects` collection in the "Admin UI > Collections > New collection".

2. In the collection panel add your project fields, eg. `title` ("text" type) and `members` ("user" type with "Max Select" = 10).

3. In the same collection panel, click on the "API rules" tab and adjust the API rules for each action (list/search, view, create, update, delete). By default all actions are "locked" and only admins can perform them. To allow only assigned users view and update access to projects, you could use for example the following API rule:
```
// PocketBase v0.7.x
@request.user.id != "" && members.id = @request.user.id

// PocketBase v0.8.0-rc
@request.auth.id != "" && members.id = @request.auth.id
```
  _There will be more documentation soon, but you could find some additional info in https://pocketbase.io/docs/manage-collections/#collection-api-rules._


4. To assign a new user to an existing project, you can do that from the Admin UI, or send a request via the JS SDK:
    ```js
    // PocketBase v0.7.x (use `create("projects", { ... })` for create)
    client.records.update("projects", "PROJECT_ID", {
        members: ["USER_ID1", "USER_ID2", "USER_ID3"],
    })
    
    // PocketBase v0.8.0-rc (use `create({ ... })` for create)
    client.collection("projects").update("PROJECT_ID", {
        members: ["USER_ID1", "USER_ID2", "USER_ID3"],
    })
    ```

One current limitation of the multi-users and multi-relations fields is that you can't just append/remove a single item from the array list and you'll need to pass it always as it is, for example if you want to remove "USER_ID2" from above, you'll have to set members to `["USER_ID1", "USER_ID3"]`.

---

_Please note that the `user` type field will be merged with the `relation` type field in the upcoming v0.8.0 release (the final release will most likely happen at the end of next week)_
